.github/workflows/python_package.yml

```diff
@@ -154,14 +176,37 @@ jobs:
           export PRODUCES_ARTIFACTS="${{ matrix.produces-artifacts }}"
           $GITHUB_WORKSPACE/.ci/setup.sh || exit 1
           $GITHUB_WORKSPACE/.ci/test.sh || exit 1
+      - name: Install OpenCL on Windows
+        if: startsWith(matrix.os, 'windows') && (matrix.task == 'bdist')
+        run: |
+          cmd /c "powershell -ExecutionPolicy Bypass -File %BUILD_SOURCESDIRECTORY%/.ci/install-opencl.ps1"
+      # 'conda init powershell' needs to be run in a separate process
+      # ref: https://docs.conda.io/projects/conda/en/stable/dev-guide/deep-dives/activation.html
+      - name: Initialize conda on Windows
+        if: startsWith(matrix.os, 'windows')
+        shell: pwsh -command ". {0}"
+        run: |
+          $env:PATH = "$env:CONDA/Scripts;$env:PATH"
+          conda init powershell
+      - name: Setup and run tests on Windows
+        if: startsWith(matrix.os, 'windows')
+        shell: pwsh -command ". {0}"
+        run: |
+          $env:COMPILER = "${{ matrix.compiler }}"
+          $env:PRODUCES_ARTIFACTS = "${{ matrix.produces-artifacts }}"
+          $env:PYTHON_VERSION = "${{ matrix.python_version }}"
+          $env:TASK = "${{ matrix.task }}"
```

Comment:
```suggestion
          $env:TASK = "${{ matrix.task }}"
          $env:METHOD = "${{ matrix.method }}"
```

Reply:
Sure we can do this. I'd omitted it because it isn't used in the test script: https://github.com/microsoft/LightGBM/pull/7086#discussion_r2563206811

I'll push a commit with it though, to preserve the alphabetical order.